[jimseah-0116] iP

jimseah-0116 opened the pull request:
## ChatBro

> "Hasta la vista, baby." — Arnold Schwarzenegger, in Terminator 2 ([source](https://youtu.be/dyt3dalRqXE?si=26olgWUW_a1XWXdX&t=481))

ChatBro helps you keep track of your tasks and fosters camaraderie, bro! It's:

 - Text-based !
 - Easy to learn !
 - ~~FAST~~ *SUPER* FAST to use !
 - For everyone, whether you're a bro, sis, whatever !

<p align="center">
  <img width="250" height="200" src="https://cdn.discordapp.com/attachments/523125126540165121/1205897803252105276/image.png?ex=65da0ad8&is=65c795d8&hm=d46972c9b87cff24cc766e2b7dac30fee232b81d78b8da4d2c64c71870e2db01&">
</p>

<br>
All you need to do is:

 1. Download it
 2. Double-click it
 3. Use the bot's commands (Type 'help' for the list of commands!)
 4. Add tasks and let ChatBro handle your tasks for you, bro 😎
<br>
<p>And, it is **FREE** !! </p>
 Features:
 
 - [x] Managing Tasks
 - [x] Manage deadlines
 - [ ] Reminders (coming soon)
 <p> Also here's some random code for fun: </p>
 

   ```
 public class Main {
        public static void main(String[] args) {
            System.out.println("Hey bro!");
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jimseah-0116 committed:
Renaming Duke to ChatBro

jimseah-0116 committed:
updated code for Level-1

jimseah-0116 committed:
updates for Level-2

jimseah-0116 committed:
updates for Level-3

jimseah-0116 committed:
updates for Level-3 and A-Classes

jimseah-0116 committed:
updates for Level-3 and A-Classes

jimseah-0116 committed:
updates for Level-4 and A-Inheritance

jimseah-0116 committed:
updates for Level-5 and A-Exceptions

jimseah-0116 committed:
edits for Level-5 for consistency

jimseah-0116 committed:
Level-6 updates

jimseah-0116 committed:
Running I/O redirection test

jimseah-0116 committed:
Task class: Move taskCount to Task class

jimseah-0116 committed:
Attempt Level-7

jimseah-0116 committed:
Add Ui class

jimseah-0116 committed:
Database.java: Add file methods

jimseah-0116 committed:
Add Database class Level-7

jimseah-0116 committed:
Bugfix for Level-7

jimseah-0116 committed:
Bugfix for Level-7

jimseah-0116 committed:
Attempt A-MoreOOP

ChuaZiLong commented on src/main/java/ChatBro.java:
```diff
@@ -0,0 +1,232 @@
+import java.util.ArrayList;
+
+public class ChatBro {
```
Good that classes are following PascalCase.

ChuaZiLong commented on src/main/java/ChatBro.java:
```diff
@@ -0,0 +1,232 @@
+        Parser parser = new Parser();
+        Database db = new Database();
+
+        ArrayList<Task> taskList = new ArrayList<>(101);
```
Perhaps include a comment explaining why size 101 was chosen would be good for understanding the code?

ChuaZiLong replied:
I like that that you are using camelCase for variable names.

ChuaZiLong commented on src/main/java/ChatBro.java:
```diff
@@ -0,0 +1,232 @@
+            String[] inputSplit = input.split(" ");
+            String command = inputSplit[0];
+            switch (command) {
+                case "list":
```
The coding standard says case should not be indented.

jimseah-0116 committed:
Improve Level-7

jimseah-0116 committed:
Improve A-MoreOOP

jimseah-0116 committed:
A-Packages

jimseah-0116 committed:
Add Level-8

jimseah-0116 committed:
Merge branch 'add-gradle-support'

jimseah-0116 committed:
A-Gradle

dwangwk commented on src/main/java/ChatBro/TaskManager.java:
```diff
@@ -0,0 +1,59 @@
+
+import java.util.ArrayList;
+
+public class TaskManager {
```
Perhaps you could add JavaDocs and comments for this class and its methods to improve readability?

dwangwk commented on src/main/java/ChatBro/ChatBro.java:
```diff
@@ -0,0 +1,33 @@
+package ChatBro;
+
+public class ChatBro {
```
Perhaps you could add some java documentation and comments to improve the readability of this class? Overall, you can also consider organising the code into smaller, more cohesive methods and classes to improve modularity and maintainability.

dwangwk commented on src/main/java/ChatBro/ChatBro.java:
```diff
@@ -0,0 +1,33 @@
+        String savedTasks = Database.readFromFile();
+        String[] savedTasksSplit = savedTasks.split("\n"); // Split savedTasks by newline
+        int length = savedTasksSplit.length;
+        for (int i = 0; i < 100; i++) { // Fill taskList with null
+            tm.getList().add(null);
+        }
```
I feel that the loop that fills the task list with null values seems unnecessary?

jimseah-0116 replied:
Task list is filled with null to prevent null pointer exception with some methods, e.g. deleteTask() method. I have added a more informative comment for this method now. Thanks for the comment!

dwangwk commented on src/main/java/ChatBro/Command.java:
```diff
@@ -0,0 +1,182 @@
+                } else if (TaskManager.getList().get(taskNum) == null) {
+                    Ui.printDoesNotExistError();
+                    return;
+                }
```
Perhaps you can consider whether any common functionality shared among commands can be abstracted into helper methods or utility classes to avoid code duplication?

dwangwk commented on src/main/java/ChatBro/Database.java:
```diff
@@ -0,0 +1,83 @@
+            String[] splitString = taskString.split("♢");
+            for (String s : splitString) {
+                if (s.isEmpty()) { // if any information is missing
+                    throw new WrongFileFormatException("savedTasks.txt is in the wrong format. Please delete the file and restart the program.");
```
Perhaps you can consider adding more specific error messages to differentiate between different types of format errors (e.g., missing fields, invalid status)?

dwangwk commented on src/main/java/ChatBro/Database.java:
```diff
@@ -0,0 +1,83 @@
+     * Parses a string into a ChatBro.Task object.
+     * @return ChatBro.Task object.
+     */
+    public static Task parseTask(String taskString) throws WrongFileFormatException {
```
Perhaps you could abstract the methods for parsing to the parser class?

dwangwk reviewed (commented):
Overall, great attempt at fulfilling the requirements for the IP. I feel that the code structure could be improved? A lot of logic can be abstracted out to separate methods to improve readability and maintainability? Perhaps you can consider creating more packages specifically for tasks and exceptions instead of one single package with all the files.

jimseah-0116 committed:
Add JUnit Test

jimseah-0116 committed:
Level-9

jimseah-0116 committed:
A-JavaDoc

jimseah-0116 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/ChatBro/Ui.java

jimseah-0116 committed:
Fix Level-8 tag

jimseah-0116 committed:
Fix A-JavaDoc tag

jimseah-0116 committed:
Try JavaFX

jimseah-0116 committed:
GUI images

jimseah-0116 committed:
Encounter bug in fxmlLoader

jimseah-0116 committed:
MainWindow class: fix initialize method

initialize method causes NullPointerException.
A fix makes the chatbot work.

Add @FXML annotation to DialogBox dialog and displayPicture fields.

jimseah-0116 committed:
Perform A-Assertion

jimseah-0116 committed:
Add more assertions

jimseah-0116 committed:
Classes don't follow coding standard.

Coding standard should be followed to improve readability and ease of understanding.

Change some parts to adhere to coding standard and improve code quality.

jimseah-0116 committed:
Merge pull request #2 from jimseah-0116/branch-A-Assertions

Branch a assertions

jimseah-0116 committed:
Merge branch 'branch-A-CodeQuality'

jimseah-0116 committed:
Add IntervalDeadline class

IntervalDeadline is a task with a deadline that is an interval of time.

jimseah-0116 committed:
Update README.md

jimseah-0116 committed:
Add Ui.png

jimseah-0116 committed:
Merge remote-tracking branch 'origin/master'

jimseah-0116 committed:
Improve code quality and coding standards

jimseah-0116 committed:
Update README.md

jimseah-0116 committed:
Update README.md

jimseah-0116 committed:
Update README.md

jimseah-0116 committed:
Remove one unnecessary comment

jimseah-0116 committed:
Merge remote-tracking branch 'origin/master'

jimseah-0116 committed:
Update User Guide

Fix .jar compatability issue

Improve list and find commands for readability

jimseah-0116 committed:
Update README.md

jimseah-0116 committed:
Update README.md

jimseah-0116 committed:
Create _config.yml

jimseah-0116 committed:
Update README.md

jimseah-0116 committed:
Update README.md

jimseah-0116 committed:
Edit README

jimseah-0116 committed:
Merge remote-tracking branch 'origin/master'

jimseah-0116 committed:
Update README.md

jimseah-0116 committed:
Parser: Clean up exception throwing

jimseah-0116 committed:
Merge remote-tracking branch 'origin/master'

jimseah-0116 committed:
Parser: Clean up exception throwing

jimseah-0116 committed:
Ui class: Fix bug involving bold and italics characters

Mac users face bug in ChatBro UI

Remove bold and italics characters, update README images

jimseah-0116 committed:
Update Ui.png

jimseah-0116 committed:
Update README.md
